src/charm.py

```diff
@@ -368,6 +388,26 @@ def _on_collect_unit_status(self, e: CollectStatusEvent):
             else:
                 e.add_status(ActiveStatus())
 
+    def _on_nginx_pebble_ready(self, _) -> None:
```

Comment:
When TLS is enabled, shouldn't we push certs into the container of nginx?

Reply:
Absolutely, good catch :) Thanks to that we were actually doing unsecured communication with workers, it seems. Updated.